cluster/src/main/java/com/hartwig/pipeline/PipelineMain.java

```diff
@@ -226,6 +236,16 @@ private PipelineCompleteEventPublisher createPublisher(final EventPublisher<Pipe
         return new PipelineCompleteEventPublisher(sourceBucket, publisher, context, arguments.useCrams());
     }
 
+    private void apiUpdate(RunApi runApi, PipelineInput input, PipelineState state, Arguments arguments, String startOrFinish){
+        if (arguments.hmfApiUrl().isPresent()) {
+            Run run = runApi.get(input.operationalReferences().orElseThrow().runId());
+            if (startOrFinish.equals(START)){
+                HmfApiStatusUpdate.start(runApi, run);
+            } else if (startOrFinish.equals(FINISH)) {
+                HmfApiStatusUpdate.finish(runApi, run, state.status());
+            }
```

Comment:
Looking at this critically for the first time I think I would dispense with the `String` constants and in fact this whole method and just call `HmfApiStatusUpdate.start` or `.finish` in the approprate place.

Also I think it would be better to have an instance of `HmfApiStatusUpdate` rather than calling static methods, I think we may have discussed this on the call we had, not sure. I'd pass the run to the constructor in that case.